### DISCUSSION TITLE
Question: will term-image work with textual?

### DISCUSSION THREAD

**Discussion by wey-gu** [AnonymouX47/term-image#99] [Q&A] [answered]
I am thinking of creating a TUI with Textual, where some of its views will generate PNG in memory and I like to render it with term-image(great work!).

Is this feasible?

ref: https://github.com/Textualize/textual

Thanks!

**Comment by AnonymouX47** [comment, marked as answer]
Hello!
Thanks for the compliments and support.

Feasible? Yes, definitely... but at the moment it'll be quite difficult to pull off, most especially for the graphics-based (high-resolution) render styles. In a couple words, the reason is... there's quite a lot that comes into play when displaying images in a TUI, unlike a CLI.

I actually have this (support for Textual) [on the roadmap](https://github.com/AnonymouX47/term-image/milestone/9). The not-so-pleasant but inevitable news is that there's a **lot** that has be put in place in order to make this happen and up to par. I'm currently working on [an almost entire redesign of the API](https://github.com/AnonymouX47/term-image/milestone/10) which started from [the last release](https://github.com/AnonymouX47/term-image/releases/tag/v0.7.0) and is ongoing at #97. See the [milestones](https://github.com/AnonymouX47/term-image/milestones) for a loose summary of the roadmap.

In case you want to ask for an ETA... unfortunately, I can't give one but rest assured that I'm putting my best efforts to it. I also want to see all the things I've conceived actualized ASAP 😃.

Once again, thank you! 😃

**Reply by wey-gu** [reply]
Wow, thanks a lot, @AnonymouX47 for the Detailed and patient explanation!

I should have given a search to know the roadmap before asking.

Thanks a lot for the great work and effort ongoing, please take your time! I couldn't imagine how to enable this towards so many different emulators/OS to render pictures natively(rather than pixels, I tried rich pixels, great work, but cannot do the high-resolution thing).

It's so very lucky to have you in the world.

Best Regards
Wey

**Reply by AnonymouX47** [reply]
It's a pleasure.

Thanks so much for the compliments and support and even more, for your understanding.

There's totally nothing wrong about you asking :smiley:. Though, I think I should put a link to the GitHub milestones on the README. :thinking:
The "Planned Features" section of the docs is quite less broad and can only be updated every release, at least, for the `stable` version which is the default. I update the GitHub milestones much for frequently and can inherently provide more information about progress. 

I hope to get back to you real soon :smiley:.

Best regards.
